src/Stratis.Bitcoin.Features.MemoryPool/Rules/CheckReplacementMempoolRule.cs

```diff
@@ -0,0 +1,151 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Extensions.Logging;
+using NBitcoin;
+using Stratis.Bitcoin.Features.MemoryPool.Interfaces;
+
+namespace Stratis.Bitcoin.Features.MemoryPool.Rules
+{
+    /// <summary>
+    /// Check if transaction can replace others.
+    /// Only transactions that increase fees over previous transactions are accepted.
+    /// There is a restriction on the maximum number of transactions that would be replaced.
+    /// The new transaction must have all inputs confirmed.
+    /// The new transaction must have sufficient fees to pay for it's bandwidth.
+    /// </summary>
+    public class CheckReplacementMempoolRule : MempoolRule
+    {
+        public CheckReplacementMempoolRule(Network network,
+            ITxMempool mempool,
+            MempoolSettings mempoolSettings,
+            ChainIndexer chainIndexer,
+            ILoggerFactory loggerFactory) : base(network, mempool, mempoolSettings, chainIndexer, loggerFactory)
+        {
+        }
+
+        public override void CheckTransaction(MempoolValidationContext context)
+        {
+            // Check if it's economically rational to mine this transaction rather
+            // than the ones it replaces.
+            context.ConflictingFees = 0;
+            context.ConflictingSize = 0;
+            context.ConflictingCount = 0;
+            context.AllConflicting = new TxMempool.SetEntries();
+
+            // If we don't hold the lock allConflicting might be incomplete; the
+            // subsequent RemoveStaged() and addUnchecked() calls don't guarantee
+            // mempool consistency for us.
+            //LOCK(pool.cs);
+            if (context.SetConflicts.Any())
+            {
+                var newFeeRate = new FeeRate(context.ModifiedFees, context.EntrySize);
+                var setConflictsParents = new List<uint256>();
+                const int MaxDescendantsToVisit = 100;
```

Comment:
Move the const towards the top of the file.

Reply:
Will do in a subsequent PR